### ISSUE TITLE
"TypeError: Subscripted generics cannot be used with class and instance checks" on 3.7

### ISSUE THREAD

**Issue by giladbarnea** [ramazanpolat/prodict#1]
Hi!
Thanks for this package, I've been looking for something exactly like this.

When initializing a Prodict class with a field that's a collection of something (eg List[str]), on python 3.7, it raises a TypeError: Subscripted generics cannot be used with class and instance checks.

This does NOT happen on python 3.6.
![20181108_135551](https://user-images.githubusercontent.com/18548473/48198028-388e1000-e360-11e8-86d8-458163dd9221.gif)

Cheers,
Gilad

**Comment by ramazanpolat** [comment]
I am currently using Python 3.6 and I can confirm that it works in 3.6 but never had a chance to test it on 3.7.
For a workaround, you can annotate your ``siblings`` attribute as ``List``, not ``List[str]``.

I will test this for 3.7 asap and get back to you.

TLDR; use this code and tell me the result.

```python
class IPerson(Prodict):
    name: str
    siblings: List
```

Keep in mind that when you annotate with a type, ``Prodict`` instantiates the attribute with that type.
For example if you annotate ``name: str``, even if you pass ``name=123``, ``Prodict`` will convert it so str with ``name = str(123)``. If you don't want to convert it but still want to have auto code completion, just use ``Any`` as type like this:

```python
class IPerson(Prodict):
    name: str
    siblings: Any
```

This way, ``Prodict`` will assign whatever you pass as ``name`` without type conversion. It will be faster and you will get auto-code completion but PyCharm won't be able to infer it's type.

**Comment by ramazanpolat** [comment]
Ok found the problem. In 3.7, ``issubclass`` built-in method does not accept subscripted generics.
This is a breaking change made in Python 3.7.
Will try to fix it as soon as I found an alternative way to check subscripted instances.

**Comment by ramazanpolat** [comment]
Looks like ``prodict`` is not the only victim. Changes made in Python 3.7 for ``typing`` module also broke Fedora / RHEL installer: [pyanaconda.dbus.typing broken by Python 3.7](https://bugzilla.redhat.com/show_bug.cgi?id=1598574)   

¯\\_(ツ)_/¯

**Comment by GbeTech** [comment]
English isn't my native language so I'm not sure I got you completely, are we waiting then for a future python version that would fix whatever is broken there?
Or you're just ok with prodict working with python == 3.6 only?
What's the plan :)

**Comment by ramazanpolat** [comment]
Hi, I decided not to fix this bug because of some reasons:

1- You can use the previous solution without any problem(``siblings: List``). It will work.
2- I happen to find another library that is far better than ``Prodict``. It is called [Pydantic](https://pydantic-docs.helpmanual.io/). Upon checking the library, I've came to a conclusion that investing time in ``Prodict`` will be wasting my time. Because ``Pydantic`` is a fantastic library that already does everything ``Prodict`` can do and much more..

This is your code, using ``Prodict``:
```python
class IPerson(Prodict):
    name: str
    siblings: List[str]
```
And this is same code, but with ``Pydantic``:

```python
class IPerson(BaseModel):
    name: str
    siblings: List[str
```

As you can see, the code is exactly same and it also works in 3.7 (I double checked).

This is good news for you. There is a slight difference between ``Prodict`` and ``Pydantic`` tho. While ``Prodict`` is derived from ``dict``, ``Pydantic`` is not. But you can use ``.dict()`` method to convert it to``dict``.

I strongly encourage you to use ``Pydantic`` instead of ``Prodict``.

Although both packages share same idea and philosophy, there is no need to have 2 different libraries for doing same thing.

Check `Pydantic`` and get back to me. If you still insist, I will try to fix the bug.

**Comment by GbeTech** [comment]
Ah, yes. Pydantic does the job.
Thanks for everything!

**Comment by ramazanpolat** [comment]
Although my initial decision was to stop developing `prodict`, somehow I have found myself using `prodict` over `pydantic`. so I updated to `prodict` to `v0.8.1` **so now it successfully runs on** `python 3.7`. 

(Now the base version requirement is `3.7+` meaning `v0.8.1` won't work in python 3.6)
